azure-spring-boot-samples/azure-active-directory-b2c-oidc-spring-boot-sample/src/main/java/sample/aad/security/WebSecurityConfiguration.java

```diff
@@ -5,43 +5,28 @@
  */
 package sample.aad.security;
 
-import com.microsoft.azure.spring.autoconfigure.b2c.AADB2CAuthorizationRequestResolver;
-import com.microsoft.azure.spring.autoconfigure.b2c.AADB2CLogoutSuccessHandler;
-import com.microsoft.azure.spring.autoconfigure.b2c.AADB2CProperties;
+import com.microsoft.azure.spring.autoconfigure.b2c.AADB2COidcLoginConfigurer;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 
 @EnableWebSecurity
 public class WebSecurityConfiguration extends WebSecurityConfigurerAdapter {
 
-    private AADB2CProperties properties;
+    private final AADB2COidcLoginConfigurer configurer;
 
-    private AADB2CLogoutSuccessHandler logoutSuccessHandler;
-
-    private AADB2CAuthorizationRequestResolver requestResolver;
-
-    public WebSecurityConfiguration(AADB2CProperties properties,
-                                    AADB2CLogoutSuccessHandler logoutSuccessHandler,
-                                    AADB2CAuthorizationRequestResolver requestResolver) {
-        this.properties = properties;
-        this.logoutSuccessHandler = logoutSuccessHandler;
-        this.requestResolver = requestResolver;
+    public WebSecurityConfiguration(AADB2COidcLoginConfigurer configurer) {
+        this.configurer = configurer;
     }
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
-                .anyRequest().authenticated()
-                .and()
-                // Expire the token from b2c side.
-                .logout().logoutSuccessHandler(logoutSuccessHandler)
+                .anyRequest()
+                .authenticated()
                 .and()
-                .oauth2Login()
-                // Customize oauth2 processing url from b2c configuration.
-                .loginProcessingUrl(properties.getLoginProcessingUrl())
-                // Customize oauth2 authorization request.
-                .authorizationEndpoint().authorizationRequestResolver(requestResolver);
+                .apply(configurer)
```

Comment:
Question: should we enable it by default when user reference our package as stated in the [doc](https://docs.spring.io/spring-security/site/docs/5.1.4.RELEASE/reference/htmlsingle/#jc-custom-dsls)?
Then user only needs to add dependency and configure `application.yml` without explicit code changes.

Reply:
In fact, I tried that but it looks not a better choose I think, with following reasons:
* The HttpConfigurer almost focus on configuration from `configure(HttpSecurity http)`, it may not works well from Properties configuration like `@EnablePropertiesConfiguration`.
* It may be some annoying when we hidden too much for developers. For convention, developers leverage spring security may write configuration code like
```java
http.authorizeRequests().anyRequest().authenticated().oauth2Login();
```
And we'd better to follow the convention like
```java
http.authorizeRequests().anyRequest().authenticated().apply(aadB2CConfigurer);
```